fitparser/src/de/parser.rs

```diff
@@ -225,97 +225,41 @@ impl FitDataMessage {
-    Float32 = 0x88,
-    Float64 = 0x89,
-    UInt8z = 0x0A,
-    UInt16z = 0x8B,
-    UInt32z = 0x8C,
-    Byte = 0x0D,
-    SInt64 = 0x8E,
-    UInt64 = 0x8F,
-    UInt64z = 0x90,
 }
 
-impl BaseType {
+impl FitBaseType {
     /// The size for fixed width numeric values, for variable Llength types it's the smallest chunk size.
+    /// FitBaseType is defined in the autogenerated profile/field_types.rs, but we add this extra size
+    /// information outside of the autogenerated code.
     fn size(&self) -> u8 {
         match *self {
-            BaseType::Enum => 1,
-            BaseType::SInt8 => 1,
-            BaseType::UInt8 => 1,
-            BaseType::SInt16 => 2,
-            BaseType::UInt16 => 2,
-            BaseType::SInt32 => 4,
-            BaseType::UInt32 => 4,
-            BaseType::String => 1,
-            BaseType::Float32 => 4,
-            BaseType::Float64 => 8,
-            BaseType::UInt8z => 1,
-            BaseType::UInt16z => 2,
-            BaseType::UInt32z => 4,
-            BaseType::Byte => 1,
-            BaseType::SInt64 => 8,
-            BaseType::UInt64 => 8,
-            BaseType::UInt64z => 8,
-        }
-    }
-}
-
-impl From<u8> for BaseType {
-    /// Check the value of the last 5 bits to determine the base type.
-    ///
-    /// Bits 5 and 6 are reserved so we don't check them and to avoid issues it's easier to
-    /// simply zero them out, 0x9f == 159 == 0b10011111. When the type can't be determined we default
-    /// to a byte vector.
-    fn from(value: u8) -> Self {
-        match value & 0x9f {
-            0x00 => BaseType::Enum,
-            0x01 => BaseType::SInt8,
-            0x02 => BaseType::UInt8,
-            0x83 => BaseType::SInt16,
-            0x84 => BaseType::UInt16,
-            0x85 => BaseType::SInt32,
-            0x86 => BaseType::UInt32,
-            0x07 => BaseType::String,
-            0x88 => BaseType::Float32,
-            0x89 => BaseType::Float64,
-            0x0A => BaseType::UInt8z,
-            0x8B => BaseType::UInt16z,
-            0x8C => BaseType::UInt32z,
-            0x8E => BaseType::SInt64,
-            0x8F => BaseType::UInt64,
-            0x90 => BaseType::UInt64z,
-            0x0D => BaseType::Byte,
-            _ => BaseType::Byte,
+            FitBaseType::Enum => 1,
+            FitBaseType::Sint8 => 1,
+            FitBaseType::Uint8 => 1,
+            FitBaseType::Sint16 => 2,
+            FitBaseType::Uint16 => 2,
+            FitBaseType::Sint32 => 4,
+            FitBaseType::Uint32 => 4,
+            FitBaseType::String => 1,
+            FitBaseType::Float32 => 4,
+            FitBaseType::Float64 => 8,
+            FitBaseType::Uint8z => 1,
+            FitBaseType::Uint16z => 2,
+            FitBaseType::Uint32z => 4,
+            FitBaseType::Byte => 1,
+            FitBaseType::Sint64 => 8,
+            FitBaseType::Uint64 => 8,
+            FitBaseType::Uint64z => 8,
+            _ => panic!("Unknown FitBaseType"),
```

Comment:
The FIT profile is extensible by 3rd parties, so what we generate from isn't the only thing that might be encountered in the wild. I think we want to keep the default fallback to `_ => BaseType::Byte,` in place to avoid panicking on what might be a valid file for the profile used by the manufacturer.

Reply:
Falling back to Byte as suggested